src/indexer.c

```diff
@@ -604,30 +604,8 @@ static void hash_partially(git_indexer *idx, const uint8_t *data, size_t size)
 
 static int write_at(git_indexer *idx, const void *data, off64_t offset, size_t size)
 {
-	git_file fd = idx->pack->mwf.fd;
-	size_t mmap_alignment;
-	size_t page_offset;
-	off64_t page_start;
-	unsigned char *map_data;
-	git_map map;
-	int error;
-
-	assert(data && size);
-
-	if ((error = git__mmap_alignment(&mmap_alignment)) < 0)
-		return error;
-
-	/* the offset needs to be at the mmap boundary for the platform */
-	page_offset = offset % mmap_alignment;
-	page_start = offset - page_offset;
-
-	if ((error = p_mmap(&map, page_offset + size, GIT_PROT_WRITE, GIT_MAP_SHARED, fd, page_start)) < 0)
-		return error;
-
-	map_data = (unsigned char *)map.data;
-	memcpy(map_data + page_offset, data, size);
-	p_munmap(&map);
-
+	if (p_pwrite(idx->pack->mwf.fd, data, size, offset) != (ssize_t)size)
```

Comment:
Also, I don't like casting a `size_t` to an `ssize_t`.  I realize that it's Probably Fine on Most Systems.  But there aren't significant guarantees about these types -- IIRC, `ssize_t` is guaranteed to support a range of `-1` to the largest chunk of data that can be written (or something equally hand waving like that).  If we take a `size_t` and call a thing that reads/writes `ssize_t`s, we should probably do it in a loop, writing as many as `SSIZE_MAX` bytes _or_ `size` at a time (whichever is smaller) until we've written all of the `size` or hit an error.

Reply:
Done